tests/python_client/testcases/test_query.py

```diff
@@ -686,6 +686,141 @@ def test_query_expr_json_contains_pagination(self, enable_dynamic_field):
             res = collection_w.query(expression, limit=limit, offset=offset)[0]
             assert len(res) == limit - offset
 
+    @pytest.mark.tags(CaseLabel.L1)
+    def test_query_expr_empty_without_limit(self):
+        """
+        target: test query with empty expression and no limit
+        method: query empty expression without setting limit
+        expected: raise error
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, True)[0]
+
+        # 2. query with no limit and no offset
+        error = {ct.err_code: 1, ct.err_msg: "empty expression should be used with limit"}
+        collection_w.query("", check_task=CheckTasks.err_res, check_items=error)
+
+        # 3. query with offset but no limit
+        collection_w.query("", offset=1, check_task=CheckTasks.err_res, check_items=error)
+
+    @pytest.mark.tags(CaseLabel.L0)
+    @pytest.mark.parametrize("limit", [10, 100, 1000])
+    @pytest.mark.parametrize("auto_id", [True, False])
+    def test_query_expr_empty(self, auto_id, limit):
+        """
+        target: test query with empty expression
+        method: query empty expression with a limit
+        expected: return topK results by order
+        """
+        # 1. initialize with data
+        collection_w, _, _, insert_ids = self.init_collection_general(prefix, True, auto_id=auto_id)[0:4]
+        exp_ids, res = insert_ids[:limit], []
+        for ids in exp_ids:
+            res.append({ct.default_int64_field_name: ids})
+
+        # 2. query with limit
+        collection_w.query("", limit=limit, check_task=CheckTasks.check_query_results, check_items={exp_res: res})
```

Comment:
what about query(limit=limit), need a case here.

Reply:
And for int primary key, query(limit=limit, offset=offset) need a case too.

Reply:
Add a new case test_query_empty. for expr is an essential param in query(), both query(limit=limit) and query(limit=limit, offset=offset) will not work.